### PR TITLE
Add IAM resources for pubsub subscriptions

### LINKED ISSUE #878: Add IAM support for pubsub subscriptions
Hi,

I would love to see the addition of a `google_pubsub_subscription_iam_policy` resource to manage IAM policies on PubSub subscription:

Following is an example of what this would look like setting an IAM policy on a pubsub subscription:

```
resource "google_service_account" "subscriber" {
  account_id   = "subscriber"
  display_name = "Subscriber"
}

resource "google_pubsub_topic" "default" {
  name = "default-topic"
}

resource "google_pubsub_subscription" "default" {
  topic = "${google_pubsub_topic.default.name}"
  name  = "default-subscription"
}

data "google_iam_policy" "subscription_subscriber" {
  binding {
    role = "roles/pubsub.subscriber"

    members = [
      "serviceAccount:${google_service_account.subscriber.email}",
    ]
  }
}

# THE FOLLOWING RESOURCE DOES NOT EXIST YET!
resource "google_pubsub_subscription_iam_policy" "default_subscription_subscriber" {
  target      = "${google_pubsub_subscription.default.name}"
  policy_data = "${data.google_iam_policy.subscription_subscriber.policy_data}"
}
```

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #878 

```sh
--- PASS: TestAccPubsubSubscriptionIamMember (7.76s)
--- PASS: TestAccPubsubSubscriptionIamPolicy (10.15s)
--- PASS: TestAccPubsubSubscriptionIamBinding (10.95s)
```

### CODE REVIEW

File: google/resource_pubsub_subscription_iam_test.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+resource "google_service_account" "test-account-2" {
+  account_id   = "%s-2"
+  display_name = "Iam Testing Account"
+}
+
+resource "google_pubsub_subscription_iam_binding" "foo" {
+  subscription = "${google_pubsub_subscription.subscription.id}"
+  role    = "roles/pubsub.subscriber"
```
Comment (danawillow) [inline google/resource_pubsub_subscription_iam_test.go]: nit: spacing seems to be off here

File: google/resource_pubsub_subscription_iam_test.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPubsubSubscriptionIamPolicy_basic(subscription, topic, account, "roles/pubsub.subscriber"),
+				Check: testAccCheckPubsubSubscriptionIam(subscription, "roles/pubsub.subscriber", []string{
+					fmt.Sprintf("serviceAccount:%s@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				Config: testAccPubsubSubscriptionIamPolicy_basic(subscription, topic, account, "roles/pubsub.subscriber"),
```
Comment (danawillow) [inline google/resource_pubsub_subscription_iam_test.go]: What's the difference between this step and the previous step?

File: website/docs/r/pubsub_subscription_iam.html.markdown [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  binding {
+    role    = "roles/editor"
+    members = [
+      "user:jane@example.com",
+    ]
+  }
+}
+
+resource "google_pubsub_subscription_iam_policy" "editor" {
+  subscription       = "your-subscription-name"
```
Comment (danawillow) [inline website/docs/r/pubsub_subscription_iam.html.markdown]: spacing

File: google/iam_pubsub_subscription.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+var IamPubsubSubscriptionSchema = map[string]*schema.Schema{
+	"subscription": &schema.Schema{
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: compareSelfLinkOrResourceName,
+	},
+	"project": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (rosbo) [reply google/iam_pubsub_subscription.go]: Done

File: google/iam_pubsub_subscription.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		return err
+	}
+
+	_, err = u.Config.clientPubsub.Projects.Subscriptions.SetIamPolicy(u.subscription, &pubsub.SetIamPolicyRequest{
+		Policy: pubsubPolicy,
+	}).Do()
+
+	if err != nil {
+		return errwrap.Wrap(fmt.Errorf("Error setting IAM policy for %s.", u.DescribeResource()), err)
```
[the remark this answers is not in the document]
Comment (rosbo) [reply google/iam_pubsub_subscription.go]: My error message includes both the error and the resource friendly name. Wrapf works if you include only the error message.

File: google/resource_pubsub_subscription.go [hunk 6]
```diff
@@ -110,15 +110,22 @@ func resourcePubsubSubscriptionCreate(d *schema.ResourceData, meta interface{})
...
-func getComputedTopicName(project string, topic string) string {
-	computed_topic_name := ""
+func getComputedTopicName(project, topic string) string {
 	match, _ := regexp.MatchString("projects\\/.*\\/topics\\/.*", topic)
 	if match {
-		computed_topic_name = topic
+		return topic
 	} else {
-		computed_topic_name = fmt.Sprintf("projects/%s/topics/%s", project, topic)
+		return fmt.Sprintf("projects/%s/topics/%s", project, topic)
```
[the remark this answers is not in the document]
Comment (rosbo) [reply google/resource_pubsub_subscription.go]: Done

File: google/resource_pubsub_subscription_iam_test.go (hunk 2 above)
Comment (rosbo) [reply google/resource_pubsub_subscription_iam_test.go]: Good catch, I changed the role for the second step.

File: google/resource_pubsub_subscription_iam_test.go (hunk 1 above)
Comment (rosbo) [reply google/resource_pubsub_subscription_iam_test.go]: Fixed the spacing throughout the file

File: website/docs/r/pubsub_subscription_iam.html.markdown (hunk 3 above)
Comment (rosbo) [reply website/docs/r/pubsub_subscription_iam.html.markdown]: Fixed

**Pull request merged**